ivadomed/utils.py

```diff
@@ -619,6 +638,121 @@ def threshold_predictions(predictions, thr=0.5):
     return thresholded_preds
 
 
+def segment_volume(model_fname, model_metadata_fname, image_fname, roi_fname=None):
+    """Segment an image.
+
+    Segment an image (image_fname) using a already trained model (model_fname) given its
+    training parameters (model_metadata_fname). If provided, a RegionOfInterest (roi_fname)
+    is used to crop the image prior to segment it.
+
+    Args:
+        model_fname (string): model filename (.pt) to use.
+        model_metadata_fname (string): Configuration file filename (.json) used for the training,
+            see https://github.com/neuropoly/ivado-medical-imaging/wiki/configuration-file
+        image_fname (string): image filename (e.g. .nii.gz) to segment.
+        roi_fname (string): Binary image filename (e.g. .nii.gz) defining a region of interest,
+            e.g. spinal cord centerline, used to crop the image prior to segment it if provided.
+    Returns:
+        nibabelObject: Object containing the soft segmentation.
+
+    """
+    # Define device
+    device = torch.device("cpu")
+
+    # Load model training config
+    with open(model_metadata_fname, "r") as fhandle:
+        context = json.load(fhandle)
+
+    # If ROI is not provided then force center cropping
+    if roi_fname is None and 'ROICrop2D' in context["transformation_validation"].keys():
+        context["transformation_validation"]['CenterCrop2D'] = context["transformation_validation"].pop('ROICrop2D')
```

Comment:
This [error](https://github.com/neuropoly/ivado-medical-imaging/pull/155#issuecomment-598593242) comes from this line --> The keys in the updated dict are not in same order: `ToTensor` is applied before `CenterCrop2D`.